certbot/main.py

```diff
@@ -215,6 +215,10 @@ def _find_cert(config, domains, certname):
               RenewableCert instance or None.
     """
     action, lineage = _find_lineage_for_domains_and_certname(config, domains, certname)
+    if action == "renew":
```

Comment:
`_find_cert` should not have the side effect of modifying `config`, towards the goal of reducing the statefulness of Certbot's architecture. If we want to know for sure whether this is a renewal or certonly in practice, we can remove the machinery that asks "looks like you have a cert with these domains already, want to renew instead?". And then we'll know for sure, because there will be a single supported way to renew vs get a new cert.